server/email_verification.go

```diff
@@ -236,6 +328,12 @@ func handleEmailVerifyFunc(verifiedTpl *template.Template, issuer url.URL, keysF
 			}
 			return
 		}
+
+		qr := cbURL.Query()
+		qr.Set("code", code)
+		qr.Set("state", ses.ClientState)
```

Comment:
yeah but this state is the one that the client generated, Dex saves this state in the session, using NewSession func. I'm reading that one.
If the user takes too long to verify the email, then maybe the state could expire, but Dex will do his job and verify the email+redirect, but it's on the client's hand to check if the state is the correct one, right?